extensions/map-layers-formats/src/GoogleMaps/GoogleMaps.ts

```diff
@@ -0,0 +1,268 @@
+    formatId: GoogleMapsMapLayerFormat.formatId,
+    url: "",
+    name,
+    properties: createPropertiesFromSessionOptions(opts ?? {mapType: "satellite", language: "en-US", region: "US:", layerTypes: ["layerRoadmap"]}),
+  };
+};
+
+/**
+* Retrieves the maximum zoom level available within a bounding rectangle.
+* @param rectangle The bounding rectangle
+* @returns The maximum zoom level available within the bounding rectangle.
+* @internal
+*/
+const getViewportInfo = async (params: ViewportInfoRequestParams): Promise<ViewportInfo | undefined>=> {
+  const {rectangle, session, key, zoom} = params;
+  const north = Angle.radiansToDegrees(rectangle.north);
+  const south = Angle.radiansToDegrees(rectangle.south);
+  const east = Angle.radiansToDegrees(rectangle.east);
+  const west = Angle.radiansToDegrees(rectangle.west);
+  const url = `https://tile.googleapis.com/tile/v1/viewport?session=${session}&key=${key}&zoom=${zoom}&north=${north}&south=${south}&east=${east}&west=${west}`;
+  const request = new Request(url, {method: "GET"});
+  const response = await fetch (request);
+  if (!response.ok) {
+    return undefined;
+  }
+  const json = await response.json();
+  return json as ViewportInfo;;
+};
+
+  /**
+ * Converts the session options to provider properties
+ * @param opts Options to create the session
+ * @internal
+*/
+const createPropertiesFromSessionOptions = (opts: CreateSessionOptions): MapLayerProviderProperties => {
+  const properties: MapLayerProviderProperties = {
+    mapType: opts.mapType,
+    language: opts.language,
+    region: opts.region,
+  }
+
+  if (opts.layerTypes !== undefined) {
+    properties.layerTypes = [...opts.layerTypes];
+  }
+
+  if (opts.scale !== undefined) {
+    properties.scale = opts.scale;
+  }
+
+  if (opts.overlay !== undefined) {
+    properties.overlay = opts.overlay;
+  }
+  return properties;
+};
+
+/**
+ * Google Maps API
+ * @beta
+*/
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const GoogleMaps = {
+/**
+ * Creates Google Maps map-layer settings.
+ * @param name Name of the layer (Defaults to "GoogleMaps")
+ * @param opts Options to create the session  (Defaults to satellite map type, English language, US region, and roadmap layer type)
+ * @beta
+*/
+  createMapLayerSettings: (name?: string, opts?: CreateSessionOptions) => {
+    return ImageMapLayerSettings.fromJSON(_internal.createMapLayerProps(name, opts));
+  },
+
+/**
+ * Creates Google Maps base layer settings.
+ * @param name Name of the layer (Defaults to "GoogleMaps")
+ * @param opts Options to create the session  (Defaults to satellite map type, English language, US region, and roadmap layer type)
+ * @beta
+*/
+  createBaseLayerSettings: (opts?: CreateSessionOptions) => {
+    return BaseMapLayerSettings.fromJSON(_internal.createMapLayerProps("GoogleMaps", opts));
+  }
+};
+
+/**
+ * Internal function export for testing purposes, do not use.
+ * @internal
+ * */
+export const _internal = {
```

Comment:
@pmconne To keep the API as simple as simple, I went for this strategy..

Reply:
Which package(s) import and use this?

Reply:
@itwin/map-layers-formats's tests

Reply:
Put your _internal functions in a separate file under src/internal/. If the tests only use ESModules, import the functions directly from @itwin/map-layers-formats/lib/esm/whatever. If the tests use both ESModules and CommonJS, export your functions from src/internal/cross-package.ts.

Reply:
Actually, some of those "internal" functions are also being used inside GoogleMaps.ts. Does your suggestion still apply?

Reply:
Yes, import them into GoogleMaps.ts from the internal file.

Reply:
I pushed some adjustments to make your tests build now. The tests are in the same package as the internal APIs, so cross-package.ts isn't needed. If you intend to make additional changes that will consume those APIs from some other package, then we'll need cross-package.ts.